Lib/test/test_clinic.py

```diff
@@ -2783,6 +2796,138 @@ def test_cloned_func_with_converter_exception_message(self):
                 func = getattr(ac_tester, name)
                 self.assertEqual(func(), name)
 
+    def test_depr_star_pos0_len1(self):
+        fn = ac_tester.deprecate_positional_pos0_len1
+        self.check_depr_star(fn,
+            fname="deprecate_positional_pos0_len1",
+            pnames="'a'",
+            ok_args=(
+                ((), {"a": None}),
+            ),
+            fail_args=(
+                ((None,), {}),
+            )
+        )
```

Comment:
It's very hard to see from the test what exactly is being tested here. As well as spelling out the regex in full in this method, what if we did something like this? I think it would make it much easier to see exactly which argspecs are being tested:

```diff
-    def check_depr_star(self, fn, *, fname, pnames, ok_args, fail_args):
-        for args, kwds in ok_args:
-            fn(*args, **kwds)
+    def check_depr_star(self, *, fname, pnames, good_calls, bad_calls):
+        for call in good_calls:
+            call()
         regex = (
             f"Passing.*positional argument.*to {fname}.*is deprecated. "
             f"Parameter.*{pnames} will become.*keyword-only parameter.*in Python 3.14"
         )
-        for args, kwds in fail_args:
+        for call in bad_calls:
             with self.assertWarnsRegex(DeprecationWarning, regex) as cm:
-                fn(*args, **kwds)
+                call()
             self.assertTrue(cm.filename.endswith("test_clinic.py"), cm.filename)
-            fn(*args, **kwds)  # No warning raised second time
+            call()  # No warning raised second time

     def test_objects_converter(self):
         with self.assertRaises(TypeError):
@@ -2798,15 +2798,11 @@ def test_cloned_func_with_converter_exception_message(self):

     def test_depr_star_pos0_len1(self):
         fn = ac_tester.deprecate_positional_pos0_len1
-        self.check_depr_star(fn,
+        self.check_depr_star(
             fname="deprecate_positional_pos0_len1",
             pnames="'a'",
-            ok_args=(
-                ((), {"a": None}),
-            ),
-            fail_args=(
-                ((None,), {}),
-            )
+            good_calls=[lambda: fn(a=None)],
+            bad_calls=[lambda: fn(None)]
         )
```

Reply:
I agree, that's better. Lambdas are nice, no matter what they say.